native_client/deepspeech.h

```diff
@@ -20,32 +20,43 @@ typedef struct ModelState ModelState;
 typedef struct StreamingState StreamingState;
 
 /**
- * @brief Stores each individual character, along with its timing information
+ * @brief Stores text of an individual token, along with its timing information
  */
-typedef struct MetadataItem {
-  /** The character generated for transcription */
-  char* character;
+typedef struct TokenMetadata {
+  /** The text corresponding to this token */
+  char* text;
```

Comment:
None of these constness changes are possible because then I can't create/assign the values in the implementation.

Reply:
Creating them should work via structure initialization. Why you need to assign to them after they are created?

Reply:
See `ModelState::decode_metadata` for example. How do you use struct initialization for a variable size array that gets allocated with new?

Reply:
It's actually even earlier than initializing them, I can't construct them:

```
ERROR: /Users/reubenmorais/Development/tensorflow/native_client/BUILD:91:1: C++ compilation of rule '//native_client:libdeepspeech.so' failed (Exit 1)
native_client/modelstate.cc:55:49: error: call to implicitly-deleted default constructor of 'TokenMetadata []'
    std::unique_ptr<TokenMetadata[]> tokens(new TokenMetadata[transcripts[i].num_tokens]);
                                                ^
native_client/deepspeech.h:27:21: note: default constructor of 'TokenMetadata' is implicitly deleted because field 'text' of const-qualified type 'const char *const' would not be initialized
  const char* const text;
```

And I can't add a constructor by hand because that would make `deepspeech.h` C++ instead of C.

Reply:
Solved!